### PR TITLE
feat: v0.5.5 — discoverability for LLM crawlers and coding agents

### LINKED ISSUE #74: Discoverability: better surface for LLM crawlers, coding agents, and search
## Goal

Make claude-status easier to discover and install for two distinct audiences:

1. **LLM crawlers / answer engines** (Perplexity, Phind, ChatGPT search, Claude.ai search, Gemini) — when a user asks "how do I add a status line to Claude Code," our project should surface as a clear, copy-paste-ready answer.

2. **Coding agents acting on a user's behalf** (Claude Code itself, Cursor, Aider, Continue, etc.) — when a user tells their agent "set up a status line for Claude Code," the agent should be able to discover this project, find the exact install command, run it, and verify success without back-and-forth.

### Crawlers / answer engines
- **Refresh \`llms.txt\`** — stale (still describes pre-v0.5.3 thresholds, says "210+ tests" when there are 284). The [llms.txt convention](https://llmstxt.org/) is increasingly consumed by LLM crawlers as a high-signal summary.
- **Tighten \`pyproject.toml\` keywords** — add terms agents/users actually search for (\`claude-code-plugin\`, \`agent-tooling\`, \`coding-agent\`).
- **GitHub repo topics** — add tags so GitHub's own search and the topic-feed surface the project.

### Coding agents
- **Add \`AGENTS.md\`** — one-page document at repo root specifically for coding agents. Includes the one-liner install command, verification step, update command, and uninstall command. The \`AGENTS.md\` convention is emerging across agent tools (Cursor rules, Aider conventions, Continue).
- **Add an \"AI / coding agents\" section to README** — short, prominent, copy-paste-ready install block near the top of the README so both human readers and crawlers see it immediately.
- **Add \`--print-config\` flag** — lets an agent introspect what's currently installed in \`~/.claude/settings.json\` without parsing the JSON itself. Useful for \"is claude-status already set up?\" agent checks.

## Out of scope

- robots.txt / sitemap (GitHub and PyPI control these; we can't override).
- Keyword stuffing or low-quality SEO content (would hurt trust on a public OSS project).

## Acceptance criteria

- \`llms.txt\` reflects v0.5.4 (correct test count, two-stage layout, correct thresholds 150/100).
- \`AGENTS.md\` exists at repo root and is reachable from the README.
- README has a top-level section visible above the fold that gives a coding agent everything it needs to install + verify.
- \`claude-status --print-config\` reports current installation status (installed/not, command, theme, refreshInterval) in a deterministic, machine-readable form, with full test coverage.
- pyproject keywords updated, repo topics added.
- All 284+ existing tests still pass; new tests for \`--print-config\`.
- CHANGELOG.md updated with the v0.5.5 entry.

### PR DESCRIPTION
**Pull request by mkalkere**
Closes #74.

## Summary

Adds discoverability improvements targeting two distinct audiences — LLM crawlers / answer engines and coding agents acting on a user's behalf — while remaining additive (no rendering behavior changes).

### For LLM crawlers (Perplexity, Phind, ChatGPT search, Claude.ai search, Gemini)
- Refreshed `llms.txt` for v0.5.4+ (current thresholds 150/100, two-stage layout description, current test count, link to AGENTS.md)
- Broadened PyPI keywords (`claude-code-plugin`, `coding-agent`, `agent-tooling`, `ai-coding`, `llm-tooling`, `ai-developer-tools`)
- Added 2 GitHub repo topics (`coding-agent`, `agent-tooling`) — at the 20-topic limit

### For coding agents (Claude Code, Cursor, Aider, Continue, Cline, …)
- New `AGENTS.md` at repo root: non-interactive install one-liner, verification, update, uninstall, themes, budget config, common recipes
- New "For Coding Agents" README section above the fold with the same one-liner
- New `--print-config` flag: machine-readable install state in 8 stable `key=value` lines, three exit codes (0/1/2), shell-script-friendly

## The `--print-config` contract

```
installed=true
command=claude-status
type=command
refreshInterval=<integer or empty>
theme=<theme name; "default" when --theme not specified>
version=<package version, e.g. 0.5.5>
settings_path=<absolute path to settings.json>
settings_state=ok
```

**Exit codes:**
- `0` installed and ready
- `1` not installed (settings missing, statusLine missing, or statusLine points at another tool)
- `2` settings.json corrupt or unreadable. **Agents must NOT auto-install on `2`** — would overwrite recoverable user config. Stderr carries a diagnostic.

**Install detection** covers every working pattern: direct binary (`claude-status`, `claude-status.exe`, full Windows paths with spaces), module form (`python -m claude_statusline`), runner forms (`uvx claude-status`, `pipx run claude-status`). Lookalikes (`not-claude-status`, `my-claude-status-fork`) correctly do not match.

## Review iterations

The initial implementation (commit 7ac78e2) had real issues that all four PR-review agents independently flagged:
- Install detection used `endswith` heuristic — missed Windows `.exe`, missed `python -m`, false-matched lookalikes
- Corrupt settings.json reported as "not installed" with exit 1, letting an agent silently overwrite recoverable config
- Newlines in stored command field could inject fake `key=value` lines, breaking the line-count contract
- `--theme=NAME` form was silently dropped → reported `theme=default`
- `refreshInterval` numeric string silently dropped (common in hand-edited settings.json)
- AGENTS.md had a wrong backup path and a stale `version=0.5.4` sample

All addressed in commit c46f139 with 20 new edge-case regression tests.

## Test plan
- [x] Full test suite passes: **315 tests** (up from 295 in v0.5.4), OK
- [x] `--print-config` output verified across 14 input scenarios with exact expected output (installed/not, themes, refreshInterval forms, lookalikes, non-dict statusLine, newline injection, etc.) — all 8 lines, exit codes 0/1/2 propagate correctly
- [x] Corrupt settings.json: exit 2, stderr diagnostic, 8 lines on stdout, `settings_state=unreadable`
- [x] No new dependencies (stdlib only — `shlex` already in stdlib)
- [x] No behavior changes to the rendered status line itself

## Files
- `claude_statusline/cli.py` — `cmd_print_config`, `_is_claude_status_invocation`, `_extract_theme`, `_sanitize_field` helpers; `--print-config` argparse flag
- `tests/test_all.py` — `TestPrintConfig` (11 tests) + `TestPrintConfigEdgeCases` (20 tests)
- `AGENTS.md` (new) — agent-facing install guide
- `llms.txt` — refreshed for v0.5.5
- `README.md` — "For Coding Agents" section + `--print-config` in the CLI table
- `CHANGELOG.md` — v0.5.5 entry
- `pyproject.toml` — version 0.5.5, broadened keywords

### CODE REVIEW

File: claude_statusline/cli.py [hunk 1]
```diff
@@ -864,6 +865,166 @@ def cmd_demo():
...
+    if not parts:
+        return False
+    head = os.path.basename(parts[0]).lower()
+    # strip .exe suffix on Windows-style invocations
+    if head.endswith(".exe"):
+        head = head[:-4]
+    if head == "claude-status":
+        return True
+    # python -m claude_statusline (or py -m / python3 -m)
+    if head in ("python", "python3", "py") and "-m" in parts:
```
[the remark this answers is not in the document]
Comment (mkalkere) [reply claude_statusline/cli.py]: Good catch — addressed in 71a6328. Replaced the hard-coded `("python", "python3", "py")` allowlist with a regex `^python(\d+(\.\d+)*)?$` (plus bare `py`). Covers `python3.11`, `python3.12.5`, etc., while still rejecting `pythonista`, `python-fork`, `ipython`, `pythonw` (which a plain `startswith("python")` would falsely accept). Added regression tests for both directions.

File: claude_statusline/cli.py [hunk 2]
```diff
@@ -864,6 +865,166 @@ def cmd_demo():
...
+    Returns "default" if no --theme was specified, "" if parts is empty.
+    """
+    if not parts:
+        return ""
+    for i, tok in enumerate(parts):
+        if tok == "--theme" and i + 1 < len(parts):
+            return parts[i + 1]
+        if tok.startswith("--theme="):
+            return tok.split("=", 1)[1]
+    return "default"
```
[the remark this answers is not in the document]
Comment (mkalkere) [reply claude_statusline/cli.py]: Right — fixed in 71a6328. Switched to last-wins to match argparse semantics, so the reported theme reflects what the running command actually uses. Tested against both forms (`--theme nord --theme focus`) and mixed (`--theme=nord --theme focus`).

File: claude_statusline/cli.py [hunk 3]
```diff
@@ -864,6 +865,166 @@ def cmd_demo():
...
+            settings = None
+            settings_state = "unreadable"
+        if isinstance(settings, dict):
+            sl = settings.get("statusLine")
+            if isinstance(sl, dict):
+                sl_type = sl.get("type", "")
+                if not isinstance(sl_type, str):
+                    sl_type = str(sl_type)
+                cmd_value = sl.get("command", "")
+                cmd_str = cmd_value if isinstance(cmd_value, str) else str(cmd_value)
```
[the remark this answers is not in the document]
Comment (mkalkere) [reply claude_statusline/cli.py]: Real footgun — fixed in 71a6328. Added explicit `None` checks before `str()` for both `command` and `type` so a literal null in settings.json emits an empty string per the documented contract instead of the string "None".
